test/pkg/broker/broker.go

```diff
@@ -33,9 +40,32 @@ func Creator(client *eventingtestlib.Client, version string) string {
 
 	switch version {
 	case "v1":
+		namespace := client.Namespace
+		// Create Broker's own ConfigMap to prevent using defaults.
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "kafka-broker-upgrade-config",
+				Namespace: namespace,
+			},
+			Data: map[string]string{
+				broker.BootstrapServersConfigMapKey:              testingpkg.BootstrapServersPlaintext,
+				broker.DefaultTopicNumPartitionConfigMapKey:      fmt.Sprintf("%d", testingpkg.NumPartitions),
+				broker.DefaultTopicReplicationFactorConfigMapKey: fmt.Sprintf("%d", testingpkg.ReplicationFactor),
+			},
+		}
+		cm, err := client.Kube.CoreV1().ConfigMaps(namespace).Create(context.Background(), cm, metav1.CreateOptions{})
+		if err != nil && !apierrors.IsAlreadyExists(err) {
+			client.T.Fatalf("Failed to create ConfigMap %s/%s: %v", namespace, cm.GetName(), err)
+		}
 		client.CreateBrokerOrFail(
 			name,
 			resources.WithBrokerClassForBroker(kafka.BrokerClass),
+			resources.WithConfigForBroker(&duckv1.KReference{
+				Kind:       "ConfigMap",
+				Namespace:  cm.GetNamespace(),
+				Name:       cm.GetName(),
```

Comment:
If `apierrors.IsAlreadyExists(err) == true`, cm at line 65 and 66 is `nil`, we should just use `namespace` and a new name variable set to `kafka-broker-upgrade-config`.